dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostFactoryImpl.java

```diff
@@ -791,6 +791,17 @@ protected Optional<List<Host>> search(final String siteNameFilter, final String
             hostList.addAll(APILocator.getPermissionAPI().filterCollection(siteList, PermissionAPI
```

Comment:
this is not on you but.. we should never return a List of Optional 
Optional is meant to represent an empty state condition which can also be accomplished by List 
You can do List.isEmpty and that should get you that behavior